doc improvement: update to Azure IoT hub cert_tool.py script (#14297)
nrfconnect/sdk-nrf: doc/nrf/libraries/networking/azure_iot_hub.rst

Generating certificates
=======================

The connection to the Azure IoT Hub with MQTT is secured using TLS.
To create the device certificate, you need a Certificate Authority (CA) certificate and a private key that is used to sign all of your client certificates.
The CA certificate is uploaded to Azure IoT Hub, so Azure can verify that the device certificate is signed by your CA.
If you do not have a CA certificate, you can purchase one or make a self-signed test CA certificate for testing purposes.

> To help generate test CA certificates and handle the device keys and certificates, you can use a Python script, which is placed in the :file:`nrf/scripts/cert_tool.py` file.

Reviewer: ```suggestion
To help generate test CA certificates and handle the device keys and certificates, you can use the :file:`nrf/scripts/cert_tool.py` Python script.
```

   .. tab:: nRF91: Script generated private key

      .. warning::
         When generating the private key on your computer, make sure to keep it secure and not to share it with anyone.

>          If the private key is compromised, the security of the device is compromised.

Reviewer 2: Please keep the sentence about compromised keys. I've asked security about their input, including this warning.

Reviewer: Not really a warning, but maybe "important".

   .. tab:: nRF70: Script generated private key

      .. warning::
         When generating the private key on your computer, make sure to keep it secure and not to share it with anyone.

>          If the private key is compromised, the security of the device is compromised.

Reviewer 2: Please keep the sentence about compromised keys. I've asked security about their input, including this warning.

The Azure IoT Hub library requires provisioning of the following certificates and a private key for a successful TLS connection:

1. `DigiCert Global Root G2`_ - The root CA certificate used to verify the server's certificate chain while connecting.

> #. `Baltimore CyberTrust Root Certificate`_ - Azure`s legacy root CA certificate needed to verify the Azure server's  that have not migrated to `DigiCert Global Root G2`_ yet.

Reviewer 2: I think we should use apostrophe here instead of backtick. Probably neutral `'` or right `’`.
```suggestion
#. `Baltimore CyberTrust Root Certificate`_ - Azure’s legacy root CA certificate needed to verify the Azure server's  that have not migrated to `DigiCert Global Root G2`_ yet.
```

> #. Public device certificate - Generated by the procedures described in REMOVED , used by Azure IoT Hub to authenticate the device.

Reviewer 2: ```suggestion
#. Public device certificate - Generated by the procedures described in <LINK TO SECTION> , used by Azure IoT Hub to authenticate the device.
```

Author: Which section should be referred to here.

Reviewer 2: It's part of the steps in `Generate device certificates`. We should replace "Public device certificate" to just "Device certificate", as I think the Public word just cause confusion.

Author: fixed.
